### ISSUE TITLE
How should we add unit tests?

### ISSUE THREAD

**Issue by tannerz28** [Nightwatch/api#11]
Document how we should implement Unit Tests in the API.

 * Do we put tests in the same directory as the controllers/services or do we put them in their own `tests/` directory?
 * How do we make sure it uses a different database than the production database?
 * Are there any frameworks that make tests "clean" and "neat" like the rest of the source code?
 * How do we track code coverage?

Make sure to document findings in the comments.

**Comment by jasonhaxstuff** [comment]
Well, I think they should be in their own `tests/` directory, and as for using a different database and code coverage, I have no idea.

**Comment by tannerz28** [comment]
I have figured it out, and will close this task.

The rundown:

For the API, the test stack will use Mocha, Chai, and Supertest.

Mocha and Chai are pretty standard in Node.js testing, from my experience.

Mocha is the test framework, Chai is an assertion library.

I added supertest so I could create a mock API instance, allowing me to test requesting the endpoints, and asserting the returned status code.

It was tricky to figure out how to test requests due to the API using IoC and dependency injection.
